cli/src/verify-pat.ts

```diff
@@ -0,0 +1,46 @@
+/********************************************************************************
+ * Copyright (c) 2020 TypeFox and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ********************************************************************************/
+
+import { Registry, RegistryOptions } from './registry';
+import { readManifest, addEnvOptions } from './util';
+
+/**
+ * Validates that a Personal Access Token can publish to a namespace.
+ */
+export async function verifyPat(options: VerifyPatOptions): Promise<void> {
+    addEnvOptions(options);
+    if (!options.pat) {
+        throw new Error("A personal access token must be given with the option '--pat'.");
+    }
+
+    if (!options.namespace) {
+      try {
+        options.namespace = (await readManifest()).publisher;
+      } catch (error) {
+        throw new Error(
+          `Unable to read the namespace's name. Please supply it as an argument or run ovsx from the extension folder.\n\n${error}`
+        );
+      }
+    }
+
+    const registry = new Registry(options);
+    const result = await registry.verifyPat(options.namespace, options.pat);
+    if (result.error) {
+        throw new Error(result.error);
```

Comment:
This will cause the CLI to exit with a non-zero status code, right? It's very important for automation scenarios.

Reply:
Yes!

![image](https://user-images.githubusercontent.com/562525/204065845-320c7fa8-561d-4da0-9db8-3608cca778f6.png)